### ISSUE TITLE
using --module-only for icc results in $UNKNOWN being defined to 'UNKNOWN'

### ISSUE THREAD

**Issue by boegel** [hpcugent/easybuild-easyblocks#776]
```
-bash-4.1$ module show icc/2015.3.187-GNU-4.9.3-2.25 
-------------------------------------------------------------------
/user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/modules/all/icc/2015.3.187-GNU-4.9.3-2.25:

module-whatis	 C and C++ compiler from Intel - Homepage: http://software.intel.com/en-us/intel-compilers/ 
conflict	 icc 
module		 load GNU/4.9.3-2.25 
prepend-path	 CPATH /user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/icc/2015.3.187-GNU-4.9.3-2.25/ipp/include 
prepend-path	 CPATH /user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/icc/2015.3.187-GNU-4.9.3-2.25/mkl/include 
prepend-path	 CPATH /user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/icc/2015.3.187-GNU-4.9.3-2.25/tbb/include 
prepend-path	 IPPROOT /user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/icc/2015.3.187-GNU-4.9.3-2.25/ipp 
prepend-path	 LD_LIBRARY_PATH /user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/icc/2015.3.187-GNU-4.9.3-2.25/lib 
prepend-path	 LD_LIBRARY_PATH /user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/icc/2015.3.187-GNU-4.9.3-2.25/ipp/lib/intel64 
prepend-path	 LD_LIBRARY_PATH /user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/icc/2015.3.187-GNU-4.9.3-2.25/mkl/lib/intel64 
prepend-path	 LD_LIBRARY_PATH /user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/icc/2015.3.187-GNU-4.9.3-2.25/tbb/lib/intel64/gcc4.4 
prepend-path	 LD_LIBRARY_PATH /user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/icc/2015.3.187-GNU-4.9.3-2.25/lib/intel64 
prepend-path	 LIBRARY_PATH /user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/icc/2015.3.187-GNU-4.9.3-2.25/lib 
prepend-path	 LIBRARY_PATH /user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/icc/2015.3.187-GNU-4.9.3-2.25/ipp/lib/intel64 
prepend-path	 LIBRARY_PATH /user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/icc/2015.3.187-GNU-4.9.3-2.25/mkl/lib/intel64 
prepend-path	 LIBRARY_PATH /user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/icc/2015.3.187-GNU-4.9.3-2.25/tbb/lib/intel64/gcc4.4 
prepend-path	 LIBRARY_PATH /user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/icc/2015.3.187-GNU-4.9.3-2.25/lib/intel64 
prepend-path	 MANPATH /user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/icc/2015.3.187-GNU-4.9.3-2.25/man 
prepend-path	 MANPATH /user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/icc/2015.3.187-GNU-4.9.3-2.25/man/en_US 
prepend-path	 PATH /user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/icc/2015.3.187-GNU-4.9.3-2.25/bin 
prepend-path	 PATH /user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/icc/2015.3.187-GNU-4.9.3-2.25/ipp/bin/intel64 
prepend-path	 TBBROOT /user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/icc/2015.3.187-GNU-4.9.3-2.25/tbb 
setenv		 EBROOTICC /user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/icc/2015.3.187-GNU-4.9.3-2.25 
setenv		 EBVERSIONICC 2015.3.187 
setenv		 EBDEVELICC /user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/icc/2015.3.187-GNU-4.9.3-2.25/easybuild/icc-2015.3.187-GNU-4.9.3-2.25-easybuild-devel 
prepend-path	 INTEL_LICENSE_FILE /user/home/gent/vsc400/vsc40023/licenses/intel/license.lic 
prepend-path	 NLSPATH /user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/icc/2015.3.187-GNU-4.9.3-2.25/idb/intel64/locale/%l_%t/%N 
-------------------------------------------------------------------
```

```
$ eb icc-2015.3.187-GNU-4.9.3-2.25.eb -df --module-only
== temporary log file in case of crash /tmp/eb-ZiUSvG/easybuild-ReAZck.log
== processing EasyBuild easyconfig /user/scratch/gent/vsc400/vsc40023/easybuild_easy_installed/lib/python2.6/site-packages/easybuild_easyconfigs-2.5.0.dev0-py2.6.egg/easybuild/easyconfigs/i/icc/icc-2015.3.187-GNU-4.9.3-2.25.eb
== building and installing icc/2015.3.187-GNU-4.9.3-2.25...
== fetching files [skipped]
== creating build dir, resetting environment...
== unpacking [skipped]
== patching [skipped]
== preparing...
== configuring [skipped]
== building [skipped]
== testing [skipped]
== installing [skipped]
== taking care of extensions [skipped]
== postprocessing [skipped]
== sanity checking [skipped]
== cleaning up [skipped]
== creating module...
== permissions [skipped]
== packaging [skipped]
== COMPLETED: Installation ended successfully
== Results of the build can be found in the log file /user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/icc/2015.3.187-GNU-4.9.3-2.25/easybuild/easybuild-icc-2015.3.187-20151214.223412.log
== Build succeeded for 1 out of 1
== Temporary log file(s) /tmp/eb-ZiUSvG/easybuild-ReAZck.log* have been removed.
== Temporary directory /tmp/eb-ZiUSvG has been removed.
```

```
$ module show icc/2015.3.187-GNU-4.9.3-2.25 
-------------------------------------------------------------------
/user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/modules/all/icc/2015.3.187-GNU-4.9.3-2.25:

module-whatis	 C and C++ compiler from Intel - Homepage: http://software.intel.com/en-us/intel-compilers/ 
conflict	 icc 
module		 load GNU/4.9.3-2.25 
prepend-path	 CPATH /user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/icc/2015.3.187-GNU-4.9.3-2.25/ipp/include 
prepend-path	 CPATH /user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/icc/2015.3.187-GNU-4.9.3-2.25/mkl/include 
prepend-path	 CPATH /user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/icc/2015.3.187-GNU-4.9.3-2.25/tbb/include 
prepend-path	 IPPROOT /user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/icc/2015.3.187-GNU-4.9.3-2.25/ipp 
prepend-path	 LD_LIBRARY_PATH /user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/icc/2015.3.187-GNU-4.9.3-2.25/lib 
prepend-path	 LD_LIBRARY_PATH /user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/icc/2015.3.187-GNU-4.9.3-2.25/ipp/lib/intel64 
prepend-path	 LD_LIBRARY_PATH /user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/icc/2015.3.187-GNU-4.9.3-2.25/mkl/lib/intel64 
prepend-path	 LD_LIBRARY_PATH /user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/icc/2015.3.187-GNU-4.9.3-2.25/tbb/lib/intel64/gcc4.4 
prepend-path	 LD_LIBRARY_PATH /user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/icc/2015.3.187-GNU-4.9.3-2.25/lib/intel64 
prepend-path	 LIBRARY_PATH /user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/icc/2015.3.187-GNU-4.9.3-2.25/lib 
prepend-path	 LIBRARY_PATH /user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/icc/2015.3.187-GNU-4.9.3-2.25/ipp/lib/intel64 
prepend-path	 LIBRARY_PATH /user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/icc/2015.3.187-GNU-4.9.3-2.25/mkl/lib/intel64 
prepend-path	 LIBRARY_PATH /user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/icc/2015.3.187-GNU-4.9.3-2.25/tbb/lib/intel64/gcc4.4 
prepend-path	 LIBRARY_PATH /user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/icc/2015.3.187-GNU-4.9.3-2.25/lib/intel64 
prepend-path	 MANPATH /user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/icc/2015.3.187-GNU-4.9.3-2.25/man 
prepend-path	 MANPATH /user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/icc/2015.3.187-GNU-4.9.3-2.25/man/en_US 
prepend-path	 PATH /user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/icc/2015.3.187-GNU-4.9.3-2.25/bin 
prepend-path	 PATH /user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/icc/2015.3.187-GNU-4.9.3-2.25/ipp/bin/intel64 
prepend-path	 TBBROOT /user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/icc/2015.3.187-GNU-4.9.3-2.25/tbb 
setenv		 EBROOTICC /user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/icc/2015.3.187-GNU-4.9.3-2.25 
setenv		 EBVERSIONICC 2015.3.187 
setenv		 EBDEVELICC /user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/icc/2015.3.187-GNU-4.9.3-2.25/easybuild/icc-2015.3.187-GNU-4.9.3-2.25-easybuild-devel 
prepend-path	 UNKNOWN UNKNOWN 
prepend-path	 NLSPATH /user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/icc/2015.3.187-GNU-4.9.3-2.25/idb/intel64/locale/%l_%t/%N 
-------------------------------------------------------------------
```

**Comment by boegel** [comment]
based on the order of thing, the definition of `$INTEL_LICENSE_FILE` is transformed into defining `$UNKNOWN`

**Comment by boegel** [comment]
mentioned on the mailing list, tagged for EB v2.9.0

we should also add a test for this once a fix is worked out

**Comment by boegel** [comment]
fixed in #994
